graylog2-web-interface/src/components/common/ReactGridContainer.tsx

```diff
@@ -0,0 +1,272 @@
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css, withTheme, DefaultTheme } from 'styled-components';
+import { Responsive, WidthProvider } from 'react-grid-layout';
+
+import { themePropTypes } from 'theme';
+import 'react-grid-layout/css/styles.css';
+import 'react-resizable/css/styles.css';
+import { WidgetPositionJSON } from 'views/logic/widgets/WidgetPosition';
+
+const WidthAdjustedReactGridLayout = WidthProvider(Responsive);
+
+const WidthProvidedGridLayout = (props: React.ComponentProps<typeof WidthAdjustedReactGridLayout> & { children: React.ReactNode }) => {
+  const { width } = props;
+
+  return width ? <Responsive {...props} /> : <WidthAdjustedReactGridLayout {...props} />;
+};
+
+WidthProvidedGridLayout.propTypes = { width: PropTypes.number };
+WidthProvidedGridLayout.defaultProps = { width: undefined };
+
+const StyledWidthProvidedGridLayout = styled(WidthProvidedGridLayout)(({ theme }) => css`
+  &.locked {
+    .widget-drag-handle {
+      display: none;
+    }
+  }
+
+  &.unlocked {
+    .react-draggable {
+      cursor: move;
+    }
+  }
+
+  .react-grid-item.react-grid-placeholder {
+    background: ${theme.colors.variant.info};
+  }
+
+  .actions {
+    cursor: default;
+  }
+`);
+
+const COLUMN_WIDTH = 175;
+const ROW_HEIGHT = 100;
+
+const COLUMNS = {
+  xxl: 12,
+  xl: 10,
+  lg: 8,
+  md: 6,
+  sm: 4,
+  xs: 2,
+};
+
+const BREAKPOINTS = {
+  xxl: COLUMN_WIDTH * COLUMNS.xxl,
+  xl: COLUMN_WIDTH * COLUMNS.xl,
+  lg: COLUMN_WIDTH * COLUMNS.lg,
+  md: COLUMN_WIDTH * COLUMNS.md,
+  sm: COLUMN_WIDTH * COLUMNS.sm,
+  xs: COLUMN_WIDTH * COLUMNS.xs,
+};
+
+const _gridClass = (locked, isResizable, draggableHandle, propsClassName) => {
+  const className = `${propsClassName}`;
+
+  if (locked || !isResizable) {
+    return `${className} locked`;
+  }
+
+  if (draggableHandle) {
+    return className;
+  }
+
+  return `${className} unlocked`;
+};
+
+const _onLayoutChange = (newLayout, onPositionsChange) => {
+  const newPositions = [];
+
+  newLayout.forEach((widget) => {
+    newPositions.push({
+      id: widget.i,
+      col: widget.x + 1,
+      row: widget.y + 1,
+      height: widget.h,
+      width: widget.w,
+    });
+  });
+
+  onPositionsChange(newPositions);
+};
+
+type Props = {
+  animate?: boolean,
+  children: React.ReactNode,
+  className?: string,
+  columns?: {
+    xxl: number,
+    xl: number,
+    lg: number,
+    md: number,
+    sm: number,
+    xs: number,
+  },
+  draggableHandle?: string,
+  isResizable?: boolean,
+  locked?: boolean,
+  measureBeforeMount?: boolean,
+  onPositionsChange: (newPositions: Array<WidgetPositionJSON>) => void,
```

Comment:
Previously we only called this function after checking if the layout changed, because `onLayoutChanged`  "may be triggered when clicking somewhere in a widget, check before propagating the change".

I could not find a case were `onLayoutChanged` will be triggered unintentional.